src/Server/Static.gren

```diff
@@ -90,91 +73,31 @@ type alias Config =
     }
 
 
-{-| Create a `Path` from an `Array String`.
--}
-pathFromArray : Array String -> Path
-pathFromArray passedPath =
-    passedPath
-        |> Array.popLast
-        |> Maybe.map
-                (\{ initial, last } ->
-                    case String.split "." last of
-                        [] ->
-                            Path passedPath Nothing
-
-                        [ item ] ->
-                            Path passedPath Nothing
-
-                        filePath ->
-                            Path initial (Just filePath)
-                )
-        |> Maybe.withDefault (Path passedPath Nothing)
-
-
-{-| Change a `Path` back into an `Array String`
--}
-arrayFromPath : Path -> Array String
-arrayFromPath passedPath =
-    case passedPath of
-        Path path (Just fileName) ->
-            let
-                constructedFileName =
-                    fileName
-                        |> Array.intersperse "."
-                        |> Array.foldr (++) ""
-            in
-            Array.pushLast constructedFileName path
-
-        Path path Nothing ->
-            path
-
-
 {-| Read a file from disk as `Bytes`. 
 -}
-readFileAsBytes : FileSystem.Permission -> String -> Task ReadFileError Bytes
+readFileAsBytes : FileSystem.Permission -> FileSystem.Path.Path -> Task FileSystem.Error Bytes
 readFileAsBytes fsPermission fileName =
-    FileSystem.openForRead fsPermission fileName
-        |> Task.mapError AccessError
-        |> Task.andThen
-                (\permission ->
-                    FileSystem.read permission
-                        |> Task.mapError UnknownFileSystemError
-                )
+    FileSystem.FileHandle.openForRead fsPermission fileName
+        |> Task.andThen FileSystem.FileHandle.read
 
 
 {-| Take the passed path (and an optional modifier), try to find a file at that path, and 
 produce a `Response` if it can be found.
 -}
-staticFileResponse : Config -> Array String -> Maybe String -> Task ReadFileError Response
-staticFileResponse { directory, httpResponse, fileSystemPermission } passedPath pathModifier =
+staticFileResponse : Config -> FileSystem.Path.Path -> Task FileSystem.Error Response
+staticFileResponse { directory, httpResponse, fileSystemPermission } path =
     let
-        {- The passed path incluing any passed modifiers
-        -}
-        path =
-            pathModifier
-                |> Maybe.map (\mod -> Array.pushLast mod passedPath)
-                |> Maybe.withDefault passedPath
-
         {- The derived mime type of the file located at `path` (if any)
         -}
+        mimeType : String
         mimeType =
-            path
-                |> Array.last
-                |> Maybe.withDefault ""
-                |> MimeType.lookup
+            MimeType.lookup path.extension
     in
-    -- Build the path from the given path and any modifiers. Leverages the `FileSystem`
-    -- built-ins to normalize it.
-    path
-        |> Array.pushFirst directory
```

Comment:
Testing this branch in prettynice and getting errors like:

```
Error "ENOENT" "ENOENT: no such file or directory, open '/main.js'
```

I'm guessing because `directory` is no longer being applied?

Reply:
This seems to fix it: https://github.com/blaix/gren-static-serve/commit/85bdd2a79232a4c75a5eeacd373695c29803067a

But I haven't tested SPA mode.